From 87658982e2672353d614b25e839177dc3526b9b5 Mon Sep 17 00:00:00 2001
From: Istio Automation <istio-testing-bot@google.com>
Date: Thu, 27 Jan 2022 01:07:13 -0800
Subject: [release-1.12] fix issue that is Helm chart creates invalid manifest
 when given boolean or numeric value for environment variables (#37025)

* add quote to env's value

* add release note

* fix release notes

Co-authored-by: ryodocx <email@ryodocx.net>
---
 manifests/charts/gateway/templates/deployment.yaml        | 2 +-
 .../istio-egress/templates/injected-deployment.yaml       | 2 +-
 .../istio-ingress/templates/injected-deployment.yaml      | 2 +-
 releasenotes/notes/36946.yaml                             | 8 ++++++++
 4 files changed, 11 insertions(+), 3 deletions(-)
 create mode 100644 releasenotes/notes/36946.yaml

diff --git a/manifests/charts/gateway/templates/deployment.yaml b/manifests/charts/gateway/templates/deployment.yaml
index b7eff9b630..b8e1f17791 100644
--- a/manifests/charts/gateway/templates/deployment.yaml
+++ b/manifests/charts/gateway/templates/deployment.yaml
@@ -77,7 +77,7 @@ spec:
           {{- end }}
           {{- range $key, $val := .Values.env }}
           - name: {{ $key }}
-            value: {{ $val }}
+            value: {{ $val | quote }}
           {{- end }}
           ports:
           - containerPort: 15090
diff --git a/manifests/charts/gateways/istio-egress/templates/injected-deployment.yaml b/manifests/charts/gateways/istio-egress/templates/injected-deployment.yaml
index 2252c8992d..1c81ab15a0 100644
--- a/manifests/charts/gateways/istio-egress/templates/injected-deployment.yaml
+++ b/manifests/charts/gateways/istio-egress/templates/injected-deployment.yaml
@@ -99,7 +99,7 @@ spec:
           {{- end }}
           {{- range $key, $val := $gateway.env }}
           - name: {{ $key }}
-            value: {{ $val }}
+            value: {{ $val | quote }}
           {{- end }}
           volumeMounts:
           {{- range $gateway.secretVolumes }}
diff --git a/manifests/charts/gateways/istio-ingress/templates/injected-deployment.yaml b/manifests/charts/gateways/istio-ingress/templates/injected-deployment.yaml
index 1115d18dd5..13e75d0e64 100644
--- a/manifests/charts/gateways/istio-ingress/templates/injected-deployment.yaml
+++ b/manifests/charts/gateways/istio-ingress/templates/injected-deployment.yaml
@@ -99,7 +99,7 @@ spec:
           {{- end }}
           {{- range $key, $val := $gateway.env }}
           - name: {{ $key }}
-            value: {{ $val }}
+            value: {{ $val | quote }}
           {{- end }}
           volumeMounts:
           {{- range $gateway.secretVolumes }}
diff --git a/releasenotes/notes/36946.yaml b/releasenotes/notes/36946.yaml
new file mode 100644
index 0000000000..780cc2f10b
--- /dev/null
+++ b/releasenotes/notes/36946.yaml
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: installation
+issue:
+  - 36946
+releaseNotes:
+  - |
+    **Fixed** Helm chart generates invalid manifest when given boolean or numeric value for environment variables.
-- 
2.31.1

